src/components/App/App.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+// import logo from '../../logo.svg';
+import './App.css';
+import Display from "../Display/Display";
+import ButtonPanel from "../ButtonPanel/ButtonPanel";
+import {calculate } from "../../logic/calculate";
+
+class App extends Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            calculator: {
+                total: null,
+                next: null,
+                operation: null
+            }
+        }
+    }
+    handleClick = (buttonName) => {
+        const calculator = calculate(this.state.calculator, buttonName);
```

Comment:
Since you are using `this.state` to set a new state here, a better way to write this is using the functional format for `this.setState:

```
this.setState(prevState => {
          const calculator = calculate(prevState.calculator, buttonName);
          return calculator;
        });
```
For a clearer explanation why this is better please see this [article](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b).